llvm/test/CodeGen/SPIRV/scfg-add-pre-headers.ll

```diff
@@ -0,0 +1,67 @@
+; RUN: llc -mtriple=spirv-unknown-unknown -O0 %s -o - | FileCheck %s
```

Comment:
Assuming this is a simple case that's already structured and validates, can you add a RUN line for `spirv-val`?

```
; RUN: %if spirv-tools %{ llc -mtriple=spirv-unknown-unknown -O0 %s -o - -filetype=obj | spirv-val %}
```

Reply:
Fixed the test to generate "valid" SPIR-V (function returned an int instead of void, etc), but I cannot add this line just yet. There are branches, and no OpSelectionMerge/OpLoopMerge yet, so this is not a valid SPIR-V module.